stdlib/public/core/StringNormalization.swift

```diff
@@ -247,47 +246,45 @@ internal func transcodeToUTF8(
   return (readIndex, writeIndex)
 }
 
-internal func fastNormalize(
-  readIndex: String.Index,
-  guts: _StringGuts,
-  outputBuffer: UnsafeMutableBufferPointer<UInt8>,
-  icuInputBuffer: UnsafeMutableBufferPointer<UInt16>,
-  icuOutputBuffer: UnsafeMutableBufferPointer<UInt16>
+internal func _fastNormalize(
+  _ readIndex: String.Index,
+  _ sourceBuffer: UnsafeBufferPointer<UInt8>,
+  _ outputBuffer: UnsafeMutableBufferPointer<UInt8>,
+  _ icuInputBuffer: UnsafeMutableBufferPointer<UInt16>,
+  _ icuOutputBuffer: UnsafeMutableBufferPointer<UInt16>
 ) -> NormalizationResult {
-  _internalInvariant(guts.isFastUTF8)
-  return guts.withFastUTF8 { sourceBuffer in
-    let sourceCount = sourceBuffer.count - readIndex.encodedOffset
-    let start = readIndex.encodedOffset
-    let rebasedSourceBuffer = UnsafeBufferPointer(rebasing: sourceBuffer[start...])
-    do {
-      let (read, filled) = fastFill(rebasedSourceBuffer, outputBuffer)
-        if filled > 0 {
-          let nextIndex = readIndex.encoded(offsetBy: read)
-          _internalInvariant(guts.isOnUnicodeScalarBoundary(nextIndex))
+  let sourceCount = sourceBuffer.count - readIndex.encodedOffset
+  let start = readIndex.encodedOffset
+  let rebasedSourceBuffer = UnsafeBufferPointer(rebasing: sourceBuffer[start...])
+  do {
```

Comment:
What does this `do` do?

Reply:
Scopes the returns for fastFill so I don't have to come up with some other name for read/filled. Also prevents me from accidentally using the wrong variable later on.